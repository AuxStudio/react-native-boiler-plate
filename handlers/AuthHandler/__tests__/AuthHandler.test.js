import React from 'react';
import renderer from 'react-test-renderer';

import { AuthHandler } from '..';

describe('AuthHandler', () => {
  let spy;
  const dispatch = jest.fn();

  it('renders with all/minimum required props', () => {
    expect(renderer.create(<AuthHandler dispatch={jest.fn()} />)).toMatchSnapshot();
  });

  it('calls getAuth on componentDidMount', () => {
    spy = jest.spyOn(AuthHandler.prototype, 'getAuth');

    renderer.create(<AuthHandler dispatch={dispatch} />);

    expect(spy).toHaveBeenCalled();
    expect(dispatch).toMatchSnapshot();
  });

  afterEach(() => {
    if (spy) {
      spy.mockClear();
    }
  });
});
