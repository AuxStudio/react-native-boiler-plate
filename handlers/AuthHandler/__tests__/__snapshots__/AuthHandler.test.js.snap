// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthHandler calls getAuth on componentDidMount 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "type": "getAuth",
      },
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": undefined,
    },
  ],
}
`;

exports[`AuthHandler renders with all/minimum required props 1`] = `null`;
